circuits/src/zk_gadgets/fixed_point.rs

```diff
@@ -101,46 +101,147 @@ pub struct MultiproverFixedPointGadget;
 impl MultiproverFixedPointGadget {
     // === Equality === //
 
-    /// Constrain a fixed point variable to equal a native field element
-    pub fn constrain_equal_integer<L, CS>(
-        lhs: &AuthenticatedFixedPointVar<L>,
-        rhs: &MpcVariable,
-        cs: &mut CS,
-    ) where
-        L: MpcLinearCombinationLike,
-        CS: MpcRandomizableConstraintSystem,
-    {
-        // Shift the integer
-        let shifted_rhs = *TWO_TO_M_SCALAR * rhs;
-        cs.constrain(lhs.repr.clone().into() - shifted_rhs);
-    }
-
     /// Constrain a fixed point variable to be equal to the given integer
     /// when ignoring the fractional part
-    pub fn constrain_equal_integer_ignore_fraction<L, CS>(
-        lhs: &AuthenticatedFixedPointVar<L>,
-        rhs: &MpcVariable,
-        fabric: &MpcFabric,
-        cs: &mut CS,
-    ) -> Result<(), ProverError>
-    where
-        L: MpcLinearCombinationLike,
-        CS: MpcRandomizableConstraintSystem,
-    {
-        // Shift the integer and take the difference
-        let shifted_rhs = *TWO_TO_M_SCALAR * rhs;
-        let diff = lhs.repr.clone().into() - shifted_rhs;
+    pub fn constrain_equal_integer_ignore_fraction(
+        lhs: FixedPointVar,
+        rhs: Variable,
+        fabric: &Fabric,
+        cs: &mut MpcPlonkCircuit,
+    ) -> Result<(), CircuitError> {
+        // Shift the rhs and subtract from the lhs
+        let zero_var = cs.zero();
+        let one_var = cs.one();
+        let one = ScalarField::one();
+
+        let lhs_minus_rhs = cs.lc(
+            &[lhs.repr, rhs, zero_var, zero_var],
+            &[one, -*TWO_TO_M_SCALAR, one, one],
+        )?;
 
         // Constrain the difference to be less than the precision on the fixed point,
         // This is effectively the same as constraining the difference to have an
         // integral component of zero
-        let shifted_precision =
-            MpcLinearCombination::from_scalar(*TWO_TO_M_SCALAR - Scalar::one(), fabric.clone());
-        MultiproverGreaterThanEqGadget::<DEFAULT_FP_PRECISION>::constrain_greater_than_eq(
-            shifted_precision,
-            diff,
-            fabric,
-            cs,
+        let diff = cs.lc(
+            &[one_var, lhs_minus_rhs, zero_var, zero_var],
+            &[*TWO_TO_M_SCALAR, -one, one, one],
+        )?;
+
+        MultiproverGreaterThanEqZeroGadget::<DEFAULT_FP_PRECISION>::constrain_greater_than_zero(
+            diff, fabric, cs,
```

Comment:
this seems to be a different constraint than what was there previously - seems like it used to be `2^m - 1 > (lhs - rhs*2^m)`, and is now `2^m - (lhs - rhs*2^m) > 0`. am i reading this wrong?

Reply:
It was `2^m - 1 >= (lhs - rhs << m)`, rearranging the terms gives:
`2^m - (lhs - rhs*2^m) - 1 >= 0`. Ours is a discrete domain so this is equivalent to the constraint:
`2^m - (lhs - rhs * 2^m) > 0` which is what we have here

Reply:
ah, i missed the `>=`, thank you